src/lib/slice.zig

```diff
@@ -452,42 +537,67 @@ pub const Slice = struct {
     }
 
     /// Reads the parent branch from git notes for this slice
-    /// TODO: This logic is not robust since it only checks the latest commit in the branch for the note.
-    /// We might need to check all notes in the branch since the target of the feature and only accept
-    /// the latest `slice-parent:` to be source of truth.
+    /// Searches through all notes in the slice namespace to find the most recent parent note
     pub fn getParentFromNotes(self: *const Slice, alloc: Allocator) !?[]const u8 {
-        const commit_oid = self.ref.target() orelse return null;
+        // Get the slice-specific notes namespace
+        const notes_namespace = try self.notesNamespace(alloc);
+        defer alloc.free(notes_namespace);
 
-        const note = LibGit.GitNote.read(self.repo, commit_oid) catch |err| {
-            log.debug("Failed to read note for slice {s}: {}", .{ self.name(), err });
+        // Create iterator for notes in this slice's namespace
+        var notes_iter = LibGit.GitNoteIterator.init(self.repo, notes_namespace) catch |err| {
+            log.debug("Failed to create notes iterator for slice {s}: {}", .{ self.name(), err });
             return null;
         };
-
-        if (note == null) {
-            return null; // No note exists
+        defer notes_iter.free();
+
+        var latest_parent: ?[]const u8 = null;
+
+        // Iterate through all notes in the namespace to find parent notes
```

Comment:
what is the iteration order for notes, oldest last? Make sure to put a comment about that